pkg/services/librarypanels/database.go

```diff
@@ -148,50 +166,109 @@ func (lps *LibraryPanelService) disconnectLibraryPanelsForDashboard(dashboardID
 	})
 }
 
-func getLibraryPanel(session *sqlstore.DBSession, uid string, orgID int64) (LibraryPanel, error) {
-	libraryPanels := make([]LibraryPanel, 0)
-	session.Table("library_panel")
-	session.Where("uid=? AND org_id=?", uid, orgID)
-	err := session.Find(&libraryPanels)
+func getLibraryPanel(session *sqlstore.DBSession, uid string, orgID int64) (LibraryPanelWithMeta, error) {
+	libraryPanels := make([]LibraryPanelWithMeta, 0)
+	sql := `SELECT
+				lp.id, lp.org_id, lp.folder_id, lp.uid, lp.name, lp.model, lp.created, lp.created_by, lp.updated, lp.updated_by
+				, 0 as can_edit
+				, (SELECT u.login from user u where u.id = lp.created_by and u.org_id = lp.org_id) as created_by_name
+				, (SELECT u.login from user u where u.id = lp.updated_by and u.org_id = lp.org_id) as updated_by_name
```

Comment:
Is there a more better way to do this SELECT without sub selects? I'm a bit rusty here and appreciate any suggestions.

Reply:
should be able to just do a join, important to get the user email so you can return the avatar image url as well, see https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/team.go#L409 for reference

Reply:
Thank you, I'll look into this.

Reply:
I couldn't figure out how to use `xorm` with two joins using the same table name so I wrote it like an SQL statement instead.